pliers/filters/video.py

```diff
@@ -36,28 +45,33 @@ def __init__(self, every=None, hertz=None, top_n=None):
         super(FrameSamplingFilter, self).__init__()
 
     def _filter(self, video):
+        if not isinstance(video, VideoStim):
+            raise TypeError('Currently, frame sampling is only supported for '
+                            'complete VideoStim inputs.')
+
         if self.every is not None:
-            new_idx = range(int(video.fps * video.clip.duration))[::self.every]
+            new_idx = range(video.n_frames)[::self.every]
         elif self.hertz is not None:
-            interval = int(video.fps / self.hertz)
-            new_idx = range(int(video.fps * video.clip.duration))[::interval]
+            interval = video.fps / float(self.hertz)
+            new_idx = np.arange(0, video.n_frames, interval).astype(int)
+            new_idx = list(new_idx)
         elif self.top_n is not None:
             import cv2
```

Comment:
Should probably use the import checking util here to make sure `cv2` is available.